src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1065,38 +1054,50 @@ def add_units(
-    units_description : str (optional)
-        Text description of the sorting table; recommended to included parameters of sorting method, curation, etc.
+    write_in_processing_module: bool (optional, defaults to False)
+        How to save the units table in the nwb file.
+        - True will save it to the processing module to serve as a historical provenance for the official table.
+        - False will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+    units_table_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_table_name must also be 'units'.
+    unit_table_description : str (optional)
+        Text description of the units table; it is recommended to include information such as the sorting method,
+        curation steps, etc.
     write_waveforms : bool (optional, defaults to false)
         if True and sorting is a spikeextractors SortingExtractor object then waveforms are added to the units table
         after writing.
     """
+    if not isinstance(nwbfile, pynwb.NWBFile):
+        raise TypeError(f"nwbfile type should be an instance of pynwb.NWBFile but got {type(nwbfile)}")
+
     if isinstance(sorting, SortingExtractor):
-        sampling_frequency = sorting.get_sampling_frequency()
-        if sampling_frequency is None:
-            raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
         checked_sorting = OldToNewSorting(oldapi_sorting_extractor=sorting)
     else:
         checked_sorting = sorting
-    unit_ids = checked_sorting.get_unit_ids()
+
     sampling_frequency = checked_sorting.get_sampling_frequency()
     if sampling_frequency is None:
         raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
-    assert write_as in [
-        "units",
-        "processing",
-    ], f"Argument write_as ({write_as}) should be one of 'units' or 'processing'!"
-    if write_as == "units":
-        assert units_name == "units", "When writing to the nwbfile.units table, the name of the table must be 'units'!"
+
+    if not write_in_processing_module:
+        units_table_name == "units"
+
+    if write_in_processing_module:
+        ecephys_mod = get_module(
+            nwbfile=nwbfile,
+            name="ecephys",
+            description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP.",
+        )
+        if units_table_name not in nwbfile.processing:
+            units_table = pynwb.misc.Units(name=units_table_name, description=unit_table_description)
+            ecephys_mod.add(units_table)
+
+        units_table = ecephys_mod[units_table_name]
+    else:
+        if nwbfile.units is None:
+            nwbfile.units = pynwb.misc.Units(name=units_table_name, description=unit_table_description)
+        units_table = nwbfile.units
```

Comment:
why not combine these two if blocks?

Reply:
How? The logic of table creation or the one of assignation?

Reply:
```python
if not x:
    do something


if x:
    do something else
else:
    # why not 'do something' here?
    do even more
```

Reply:
I.e., why not just `nwbfile.units = pynwb.misc.Units(name="units", description=unit_table_description)` at 1098? Saves you three lines, just as readable

Reply:
Right. I get you.